lib/linter/apply-disable-directives.js

```diff
@@ -199,6 +199,81 @@ function processUnusedDisableDirectives(allDirectives) {
     );
 }
 
+/**
+ * Collect eslint-enable comments that are used after removing the suppression by eslint-disable comments.
+ * @param {Directive[]} directives The directives to check.
+ * @returns {Set<Directive>} The used eslint-enable comments
+ */
+function collectUsedEnableDirectives(directives) {
+    const withoutRuleIdKey = Symbol("withoutRuleId");
+    const enabledRules = new Map();
+
+    const usedEnableDirectives = new Set();
+
+    for (let index = directives.length - 1; index >= 0; index--) {
+        const directive = directives[index];
+
+        if (directive.type === "disable") {
+            if (enabledRules.size === 0) {
+                continue;
+            }
+            if (directive.ruleId === null) {
+
+                // If encounter `eslint-disable` without ruleId,
+                // mark all `eslint-enable` currently held in enabledRules as used.
+                // e.g.
+                //    /* eslint-disable */ <- current directive
+                //    /* eslint-enable rule-id1 */ <- used
+                //    /* eslint-enable rule-id2 */ <- used
+                //    /* eslint-enable */ <- used
+                for (const enableDirective of enabledRules.values()) {
+                    usedEnableDirectives.add(enableDirective);
+                }
+                enabledRules.clear();
+            } else {
+                const enableDirective = enabledRules.get(directive.ruleId);
+
+                if (enableDirective) {
+
+                    // If encounter `eslint-disable` with ruleId,
+                    // mark it as used if there is an `eslint-enable` with the same ruleId in enabledRules.
```

Comment:
This doesn't seem to match up with the code. If "it" refers to the disable directive, I don't see where it's being marked as used. Is `usedEnableDirectives` also used to track disable directives?

Reply:
I thought "it" was referring to "an `eslint-enable` with the same ruleId" 😓. I'll change that comment.